pkg/controller/operator/seed/reconciler.go

```diff
@@ -110,29 +110,13 @@ func (r *Reconciler) reconcile(ctx context.Context, log *zap.SugaredLogger, seed
 	seedRecorder := r.seedRecorders[seedName]
 
 	// find the owning KubermaticConfiguration
-	configList := &operatorv1alpha1.KubermaticConfigurationList{}
-	listOpts := &ctrlruntimeclient.ListOptions{
-		Namespace: r.namespace,
-	}
-
-	if err := r.masterClient.List(ctx, configList, listOpts); err != nil {
-		return fmt.Errorf("failed to find KubermaticConfigurations: %v", err)
-	}
-
-	if len(configList.Items) == 0 {
-		log.Debug("ignoring request for namespace without KubermaticConfiguration")
-		return nil
-	}
-
-	if len(configList.Items) > 1 {
-		log.Infow("there are multiple KubermaticConfiguration objects, cannot reconcile", "namespace", r.namespace)
-		return nil
+	config, err := getKubermaticConfigurationForNamespace(ctx, r.masterClient, r.namespace, log)
+	if err != nil || config == nil {
```

Comment:
Tiny nit: I would split this into two `if`s and write some descriptive error message for `if config == nil`. Currently, if err is nil, but config is also nil, it's just going to return nil, which doesn't mean much to someone reading the logs. I guess this case shouldn't be common, but it might be helpful if it ever happens.

Reply:
So I was gonna rewrite the section:

```go
// find the owning KubermaticConfiguration
config, err := getKubermaticConfigurationForNamespace(ctx, r.masterClient, r.namespace, log)
if err != nil {
	return err
}
if config == nil {
	return fmt.Errorf("no KubermaticConfiguration found in namespace %q", r.namespace)
}
```

But then I checked what `getKubermaticConfigurationForNamespace` does:

```go
if len(configList.Items) == 0 {
	log.Debug("ignoring request for namespace without KubermaticConfiguration")
	return nil, nil
}

if len(configList.Items) > 1 {
	log.Infow("there are multiple KubermaticConfiguration objects, cannot reconcile", "namespace", namespace)
	return nil, nil
}
```

So there is already a note being issued, which is why we do not need to also return an error. So I changed it to

```go
// find the owning KubermaticConfiguration
config, err := getKubermaticConfigurationForNamespace(ctx, r.masterClient, r.namespace, log)
if err != nil {
	return err
}
if config == nil {
	return nil
}
```

But when the 2nd `if` branch is used, `err` must be `nil` because of the statement above. So we can combine the two and end up with what I wrote originally :-D